fuel-core/src/args.rs

```diff
@@ -137,3 +178,34 @@ impl Opt {
         })
     }
 }
+
+pub fn dump_snapshot(path: PathBuf, _config: ChainConfig) -> anyhow::Result<()> {
+    #[cfg(not(feature = "rocksdb"))]
+    {
+        Err(anyhow!(
+            "Rocksdb must be enabled to use the database at {}",
+            path.display()
+        ))
+    }
+
+    #[cfg(feature = "rocksdb")]
+    {
+        let db = Database::open(&path)?;
+
+        let state_conf = StateConfig::generate_state_config(db);
+
+        let chain_conf = ChainConfig {
+            chain_name: _config.chain_name,
+            block_production: _config.block_production,
+            initial_state: Some(state_conf),
+            transaction_parameters: _config.transaction_parameters,
+        };
+
+        let serialized = serde_json::to_string(&chain_conf).unwrap();
+        let mut stdout = io::stdout().lock();
+
+        stdout.write_all(serialized.as_bytes())?;
```

Comment:
1. No need to unwrap here when we have the convenience of anyhow available
2. Consider using streaming serialization instead of copying the whole string into memory

```suggestion
        let mut stdout = io::stdout().lock();
        let serialized = serde_json::to_writer(stdout, &chain_conf).context("failed to dump snapshot to JSON")?;
```

Reply:
I added this in but not commit suggestion because it actually further can have some changes such to make this look like
```rust
        let stdout = io::stdout().lock();
        serde_json::to_writer(stdout, &chain_conf).context("failed to dump snapshot to JSON")?;
        ```